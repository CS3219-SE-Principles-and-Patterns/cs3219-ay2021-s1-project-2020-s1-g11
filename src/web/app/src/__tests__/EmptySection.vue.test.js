import {createLocalVue, shallowMount} from '@vue/test-utils'
import ElementUI from 'element-ui';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import EmptySection from "./../components/emptyStates/EmptySection.vue";

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);
localVue.use(ElementUI);

describe("EmptySection.vue", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        isPageLoading: false,
        userInfo: {
          isApiError: false,
        },
        dbMetaData: {
          entitiesStatus: {
            isLoading: false,
          }
        }
      }
    })
  });

  it('renders correctly', () => {
    const wrapper = shallowMount(EmptySection, {store, localVue})
    expect(wrapper.element).toMatchSnapshot()
  });
});