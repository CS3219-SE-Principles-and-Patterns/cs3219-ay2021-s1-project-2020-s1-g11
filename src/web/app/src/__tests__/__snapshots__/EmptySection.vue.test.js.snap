// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptySection.vue renders correctly 1`] = `
<div
  class="emptySection"
>
  <el-row-stub
    align="top"
    justify="start"
    tag="div"
  >
    <el-col-stub
      span="24"
      tag="div"
    >
      <div
        class="icon"
      >
        <img
          src="@/assets/businessman.png"
        />
      </div>
    </el-col-stub>
  </el-row-stub>
   
  <el-row-stub
    align="top"
    justify="start"
    tag="div"
  >
    <el-col-stub
      span="24"
      tag="div"
    >
      <p>
        <i>
           No section found. Want to visualise data? 
        </i>
      </p>
       
      <p>
        
                Let's get started by adding a new section on the right side of the panel.
            
      </p>
    </el-col-stub>
  </el-row-stub>
</div>
`;
